services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -319,18 +338,19 @@ async def get_projects_recursively_only_if_user_is_owner(
 
         result = await conn.execute(query)
 
-        rows = await result.fetchall() or []
-        results = [ProjectID(row[0]) for row in rows]
-        return results
+        rows = result.fetchall() or []
```

Comment:
Uhm, this looks suspicious to me. It goes through the cursor and blocking on it. Are you sure about this?
I would really expect this to be an awaitable. Is there not an awaitable version of it?

Reply:
@GitHK this is the new asyncpg and not aiopg engine anymore. Please read about the new engine integration in sqlalchemy 2 in https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html

Reply:
There is also an async version https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncResult.fetchall

Reply:
Discussed this offline:
Apparently `connection.execute` has to be replaced with `connection.stream` then you get an `AsynResult` which exposes an awaitable interface of `Result`

Reply:
yes @pcrespov I think this stream is used in the catalog if you need some examples

Reply:
Done.